board.cpp

```diff
@@ -0,0 +1,190 @@
+#include "board.h"
+
+const std::vector<CreateFn> Board::pieceNames = { &Create<Square>, &Create<TBlock>, &Create<LBlockL>, &Create<LBlockR>, &Create<Straight>, &Create<ZBlock>, &Create<SBlock>};
+
+Board::Board() {
+	this->board = std::vector<int>(BOARD::BOARD_SIZE, 0);
+	this->blocksPerRow = std::vector<int>(BOARD::COLUMN_SIZE, 0);
```

Comment:
if this is blocksPerRow why is the init to BOARD::COLUMN_SIZE shouldnt it be BOARD::ROW_SIZE?

Reply:
Because there are BOARD::COLUMN_SIZE rows.